stakingModel/contracts/AbstractGoodStaking.sol

```diff
@@ -8,15 +8,35 @@ pragma solidity 0.5.4;
  */
 contract AbstractGoodStaking {
 
+    /**
+     * @dev Structure to store staking details.
+     * It contains amount of tokens staked and blocknumber at which last staked.
+     */
     struct Staker {
         uint256 stakedToken;
         uint256 lastStake;
     }
 
+    /**
+     * @dev Mapping to store staking details for each user.
+     */
     mapping(address => Staker) public stakers;
 
+    /**
+     * @dev Emitted when `staker` stake `value` tokens
+     */
     event Staked(address indexed staker, uint256 value);
+
+    /**
+     * @dev Emitted when `staker` withdraws their stake `value` tokens from 
+     * `actual` contract balance.
+     */
     event StakeWithdraw(address indexed staker, uint256 value, uint256 actual);
+
+    /**
+     * @dev Emitted when fundmanager transfers intrest collected from defi protrocol.
+     * `recipient` will receive `intrestTokenValue` as intrest.
+     */
     event InterestCollected(
         address recipient,
         uint256 intrestTokenValue,
```

Comment:
add token

Reply:
ok, will update in next commit

Reply:
only token or should i add intrest token as well? @sirpy

Reply:
yeah both. also need to add ERC20 token

Reply:
ok, will update in next commit